great_expectations/core/expectation_suite.py

```diff
@@ -174,8 +184,30 @@ def add_expectation(self, expectation: Expectation) -> Expectation:
                     raise exc
 
         expectation.register_save_callback(save_callback=self._save_expectation)
+
+        self._submit_expectation_created_event(expectation=expectation)
+
         return expectation
 
+    def _submit_expectation_created_event(self, expectation: Expectation) -> None:
+        if expectation.__module__.startswith("great_expectations."):
+            custom_exp_type = False
+            expectation_type = expectation.expectation_type
+        else:
+            custom_exp_type = True
+            anonymizer = Anonymizer()
+            expectation_type = anonymizer.anonymize(expectation.expectation_type)
+
+        submit_event(
+            event=ExpectationSuiteExpectationCreatedEvent(
+                action=EXPECTATION_SUITE_EXPECTATION_CREATED,
+                expectation_id=expectation.id,
+                expectation_suite_id=self.ge_cloud_id,
+                expectation_type=expectation_type,
+                custom_exp_type=custom_exp_type,
+            )
+        )
```

Comment:
Who owns this custom expectation identification logic and the related anonymization? Perhaps a separate Anonymizer?